mne_nirs/io/snirf/_snirf.py

```diff
@@ -23,16 +23,28 @@ def write_raw_snirf(raw, fname, add_montage=False):
     Parameters
     ----------
     raw : instance of Raw
-        Data to write to file. Must contain only `fnirs_cw_amplitude` type.
+        Data to write to file. Can be either continuous wave
+        (`fnirs_cw_amplitude`) or processed (`fnirs_od`).
     fname : str
         Path to the SNIRF data file.
     add_montage : bool
         Should the montage be added as landmarks to
         facilitate compatibility with AtlasViewer.
     """
 
-    picks = _picks_to_idx(raw.info, 'fnirs_cw_amplitude', exclude=[])
-    assert len(picks) == len(raw.ch_names), 'Data must be fnirs_cw_amplitude'
+    if 'fnirs_cw_amplitude' in raw:
+        picks = _picks_to_idx(raw.info, 'fnirs_cw_amplitude', exclude=[])
+        assert len(picks) == len(raw.ch_names), \
+            'Data must be fnirs_cw_amplitude'
+        assert 'fnirs_od' not in raw, \
+            'Data must be either raw or processed'
+    elif 'fnirs_od' in raw:
+        picks = _picks_to_idx(raw.info, 'fnirs_od', exclude=[])
+        assert len(picks) == len(raw.ch_names), 'Data must be fnirs_od'
+        assert 'fnirs_cw_amplitude' not in raw, \
+            'Data must be either raw or processed'
+    else:
+        raise ValueError('Data must be fnirs_cw_amplitude or fnirs_od')
```

Comment:
I believe this will have the same effect and save writing a test for the else case.
```suggestion
    picks = _picks_to_idx(raw.info, ['fnirs_cw_amplitude', 'fnirs_od'], exclude=[])
    assert len(picks) == len(raw.ch_names),\
        'Data must only be of type fnirs_cw_amplitude or fnirs_od'
    assert len(np.unique(raw.info.get_channel_types())) == 1,\
        'All channels must be of the same type'
```